examples/step-90/step-90.cc

```diff
@@ -0,0 +1,1271 @@
+            NonMatching::LocationToLevelSet::intersected &&
+          cell->is_locally_owned())
+        {
+          double cell_L2_error_sqr = 0.0;
+          double cell_H1_error_sqr = 0.0;
+
+          copy_data.reinit(cell);
+          scratch_data.non_matching_fe_values.reinit(cell);
+
+          const std::optional<NonMatching::FEImmersedSurfaceValues<dim>>
+            &surface_fe_values =
+              scratch_data.non_matching_fe_values.get_surface_fe_values();
+          const std::vector<double> &JxW_surface =
+            surface_fe_values->get_JxW_values();
+          int n_q_points = surface_fe_values->n_quadrature_points;
+
+          std::vector<double> sol(n_q_points);
+          surface_fe_values->get_function_values(locally_relevant_solution,
+                                                 sol);
+
+          std::vector<Tensor<1, dim>> sol_grad(n_q_points);
+          surface_fe_values->get_function_gradients(locally_relevant_solution,
+                                                    sol_grad);
+
+          for (const unsigned int q :
+               surface_fe_values->quadrature_point_indices())
+            {
+              const Point<dim> &point = surface_fe_values->quadrature_point(q);
+              const Tensor<1, dim> &normal =
+                surface_fe_values->normal_vector(q);
+              const double error_at_point =
+                sol.at(q) - analytical_solution.value(point);
+              const Tensor<1, dim> vector_error_at_point =
+                (sol_grad.at(q) - (normal * sol_grad.at(q)) * normal -
+                 analytical_solution.gradient(point));
+
+              cell_L2_error_sqr +=
+                Utilities::pow(error_at_point, 2) * JxW_surface[q];
+              cell_H1_error_sqr +=
+                vector_error_at_point * vector_error_at_point * JxW_surface[q];
+            }
+          copy_data.cell_L2_error_sqr = cell_L2_error_sqr;
+          copy_data.cell_H1_error_sqr = cell_H1_error_sqr;
+
+          if (stabilization_scheme.needs_cell_worker())
+            {
+              typename DoFHandler<dim>::active_cell_iterator level_set_cell(
+                &(triangulation),
+                cell->level(),
+                cell->index(),
+                &level_set_dof_handler);
+              scratch_data.fe_values.reinit(cell);
+              scratch_data.level_set_fe_values.reinit(level_set_cell);
+              stabilization_scheme.evaluate_cell_worker(
+                locally_relevant_solution,
+                level_set,
+                cell,
+                scratch_data,
+                copy_data);
+            }
+        }
+    };
+
+    const auto copier = [&](const auto &copy_data) {
+      if (copy_data.cell_index < activeFE_indicator.size())
+        {
+          error_L2_sqr += copy_data.cell_L2_error_sqr;
+          error_H1_sqr += copy_data.cell_H1_error_sqr;
+          error_stab_sqr += copy_data.cell_stab_sqr;
```

Comment:
You copy into the member variables here. But if you wanted to put things into the convergence table at the end of this function already, you'd just declare these as function-local variables that you capture by reference in the lambda function.